File: Daily_Challenges/2043.py
class Bank:

    def __init__(self, balance: List[int]):
        self.bank = {}
        for i,money in enumerate(balance,start=1):
            self.bank[i] = money

    def transfer(self, account1: int, account2: int, money: int) -> bool:
        if account1 not in self.bank or account2 not in self.bank: return False
        if self.bank[account1]<money: return False
        self.bank[account1]-=money
        self.bank[account2]+=money
        return True

    def deposit(self, account: int, money: int) -> bool:
        if account not in self.bank:return False
        self.bank[account]+=money
        return True

    def withdraw(self, account: int, money: int) -> bool:
        if account not in self.bank or self.bank[account]< money: return False
        self.bank[account]-=money
        return True
        

File: Daily_Challenges/2043.cs
public class Bank {

    private long[] bal;
    private int n;

    public Bank(long[] balance) {
        bal = balance;
        n = balance.Length;
    }

    public bool isvalid(int account){
        return account>=1 && account<=n;
    }
    
    public bool Transfer(int account1, int account2, long money) {
        if(!isvalid(account1) || !isvalid(account2) || bal[account1-1]<money)
            return false;
        bal[account1-1]-=money;
        bal[account2-1]+=money;   
        return true;
    }
    public bool Deposit(int account, long money) {
        if(!isvalid(account)) return false;
        bal[account-1]+=money;
        return true;
    }
    
    public bool Withdraw(int account, long money) {
        if(!isvalid(account) || bal[account-1]<money) return false;
        bal[account-1]-=money;
        return true;
    }
}
